string-manipulation-3/src/reverse-words.js

```diff
@@ -0,0 +1,26 @@
+/* exported reverseWords */
+
+/* - make storage for words that will be have their contiguous (next or together in sequence)
+     string of space characters be replaced with an empty string
+   - make storage space for resulting string
+   - for each letter in words
+        - if counter does not equal the length of the words - 1
+                - split and reverse word and place into one string
+                  and append a space
+        - else counter does equal the length of the words - 1
+                - split and reverse word and place into
+                  one string
+   - give back resulting string */
+
+function reverseWords(string) {
+  var words = string.match(/\S+/g);
```

Comment:
LOTS of regex going on here 👀. Remember that these are meant to be JavaScript practice algorithms, not regex practice. Regex is definitely the most efficient way to solve these problems, but it doesn't make for great practice of logic in JavaScript.